src/main/java/org/candlepin/resource/RulesResource.java

```diff
@@ -54,35 +52,6 @@ public RulesResource(RulesCurator rulesCurator,
         this.jsProvider = jsProvider;
     }
 
-    @Override
-    public String uploadRules(String rulesBuffer) {
-        if (rulesBuffer == null || rulesBuffer.isEmpty()) {
-            log.error("Rules file is empty");
-            throw new BadRequestException(i18n.tr("Rules file is empty"));
-        }
-
-        Rules rules = null;
-
-        try {
-            String decoded = new String(Base64.decodeBase64(rulesBuffer));
-            rules = new Rules(decoded);
-        }
-        catch (Throwable t) {
-            log.error("Exception in rules upload", t);
-            throw new BadRequestException(
-                i18n.tr("Error decoding the rules. The text should be base 64 encoded"));
-        }
-
-        Rules oldRules = rulesCurator.getRules();
-        rulesCurator.update(rules);
-        sink.emitRulesModified(oldRules, rules);
```

Comment:
I think we should remove supporting bits that are now left unused as well. Such as things from EventSink and RulesCurator.

Reply:
Both are still used by the RulesImporter in the Manifest import